### PR TITLE
GEODE-3096: Unify gfsh Http clients

### PR DESCRIPTION
**Pull request by jinmeiliao**
This is just a code cleanup to consolidate the way we make httprequest in our HttpInvoker, 

before, the initial Ping request and the scheduled Ping request each would create a new SimpleHttpRequester which is different from the HttpRequester used by the HttpInvoker, now all these three uses the same HttpRequester to issue http calls.

### CODE REVIEW

**Comment by pdxrunner** [comment]
Is there a JIRA associated with this PR?

**Comment by jinmeiliao** [comment]
Yes, GEODE-3096: Refactor and unify GFSH http clients

**Comment by jinmeiliao** [comment]
yes, GEODE-3096: Refactor and unify GFSH http clients

**Comment by jinmeiliao** [comment]
precheckin green

File: geode-core/src/test/java/org/apache/geode/management/internal/web/http/support/HttpRequesterTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    verify(requester).addHeaderValues(headerCaptor.capture());
+
+    HttpHeaders headers = headerCaptor.getValue();
+    assertThat(headers.get("user").get(0)).isEqualTo("me");
+    assertThat(headers.get("password").get(0)).isEqualTo("secret");
+  }
+
+  @Test
+  public void authenticateError() throws Exception {
+
```
Comment (jdeppe-pivotal) [inline geode-core/src/test/java/org/apache/geode/management/internal/web/http/support/HttpRequesterTest.java]: Can you add a test or remove this method if it's not relevant?

Comment (jinmeiliao) [reply geode-core/src/test/java/org/apache/geode/management/internal/web/http/support/HttpRequesterTest.java]: deleted.

**Pull request merged**